diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 271d8ffbb5b7b..18073719f9e95 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -284,8 +284,20 @@ if (current_toolchain == default_toolchain) {
     srcjar_deps = [ ":chrome_android_java_google_api_keys_srcjar" ]
   }
 
+  android_aar_prebuilt("cprocsp_base_java") {
+    aar_path = "//third_party/cprocsp/libs/csp-base.aar"
+    info_path = "//third_party/cprocsp/libs/csp-base.info"
+  }
+
+  android_aar_prebuilt("cprocsp_gui_java") {
+    aar_path = "//third_party/cprocsp/libs/csp-gui.aar"
+    info_path = "//third_party/cprocsp/libs/csp-gui.info"
+  }
+
   android_library("chrome_java") {
     deps = [
+      ":cprocsp_base_java",
+      ":cprocsp_gui_java",
       ":base_module_java",
       ":chrome_app_java_resources",
       ":chrome_public_apk_template_resources",
diff --git a/chrome/android/chrome_public_apk_tmpl.gni b/chrome/android/chrome_public_apk_tmpl.gni
index 54c50b54bb7ca..0f00e512bc425 100644
--- a/chrome/android/chrome_public_apk_tmpl.gni
+++ b/chrome/android/chrome_public_apk_tmpl.gni
@@ -729,6 +729,15 @@ template("chrome_common_apk_or_module_tmpl") {
       }
     }
 
+    deps += [
+      "//chrome/android:cprocsp_base_java",
+      "//chrome/android:cprocsp_gui_java"
+    ]
+    loadable_modules += [
+      "//third_party/cprocsp/jniLibs/arm64-v8a/libcspjni.so",
+      "//third_party/cprocsp/jniLibs/arm64-v8a/libsupport.so"
+    ]
+
     forward_variables_from(invoker,
                            "*",
                            TESTONLY_AND_VISIBILITY + [
diff --git a/chrome/android/java/res_chromium_base/values/channel_constants.xml b/chrome/android/java/res_chromium_base/values/channel_constants.xml
index cc1f2a30cb3c5..3953c9ad3800f 100644
--- a/chrome/android/java/res_chromium_base/values/channel_constants.xml
+++ b/chrome/android/java/res_chromium_base/values/channel_constants.xml
@@ -7,8 +7,8 @@ found in the LICENSE file.
 
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
     <!-- The application name displayed to the user. -->
-    <string name="app_name" translatable="false">Chromium</string>
-    <string name="bookmark_widget_title" translatable="false">Chromium bookmarks</string>
-    <string name="search_widget_title" translatable="false">Chromium search</string>
-    <string name="quick_action_search_widget_title" translatable="false">Chromium quick action search</string>
+    <string name="app_name" translatable="false">Chromium-Gost</string>
+    <string name="bookmark_widget_title" translatable="false">Chromium-Gost bookmarks</string>
+    <string name="search_widget_title" translatable="false">Chromium-Gost search</string>
+    <string name="quick_action_search_widget_title" translatable="false">Chromium-Gost quick action search</string>
 </resources>
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
index 74e852c79190b..2f5afed5e07f2 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedActivity.java
@@ -252,11 +252,22 @@ import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+import ru.CryptoPro.JCSP.NCSPConfig;
+
 /**
  * This is the main activity for ChromeMobile when not running in document mode.  All the tabs
  * are accessible via a chrome specific tab switching UI.
  */
 public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent> {
+    private static final String APP_LOGGER_TAG = "CSP";
+
+    private boolean initCSPProviders() {
+        int initCode   = NCSPConfig.init(this);
+        boolean initOk = initCode == NCSPConfig.CSP_INIT_OK;
+
+        return initOk;
+    }
+
     private static final String TAG = "ChromeTabbedActivity";
 
     protected static final String WINDOW_INDEX = "window_index";
@@ -525,6 +536,9 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
                     }
                     minimizeAppAndCloseTabOnBackPress(getActivityTab());
                 });
+	if (!initCSPProviders()) {
+            Log.i(APP_LOGGER_TAG, "Couldn't initialize CSP.");
+        }
     }
 
     @Override
diff --git a/chrome/browser/search_engines/ui_thread_search_terms_data_android.cc b/chrome/browser/search_engines/ui_thread_search_terms_data_android.cc
index 1278f9e64ed1f..caf77c47e85b9 100644
--- a/chrome/browser/search_engines/ui_thread_search_terms_data_android.cc
+++ b/chrome/browser/search_engines/ui_thread_search_terms_data_android.cc
@@ -31,7 +31,7 @@ std::string UIThreadSearchTermsData::GetSearchClient() const {
 }
 
 std::string UIThreadSearchTermsData::GetYandexReferralID() const {
-  return LocaleManager::GetYandexReferralID();
+  return "5476077";
 }
 
 std::string UIThreadSearchTermsData::GetMailRUReferralID() const {
